src/index.rs

```diff
@@ -373,9 +376,14 @@ impl Index {
       path,
       started: Utc::now(),
       unrecoverably_reorged: AtomicBool::new(false),
+      event_sender: None,
     })
   }
 
+  pub fn set_event_sender(&mut self, sender: tokio::sync::mpsc::Sender<Event>) {
```

Comment:
Instead of having a setter, let's have two `open` functions, `Index::open` and `Index::open_with_event_sender`, `Index::open` can call `Index::open_with_event_sender` with None.